pkg/kubectl/cmd/delete.go

```diff
@@ -116,6 +116,7 @@ func RunDelete(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []str
 		SelectAllParam(deleteAll).
 		ResourceTypeOrNameArgs(false, args...).RequireObject(false).
 		Flatten().
+		Latest().
```

Comment:
Yes.

And I take back that overhaul the grace period mechanism will solve this race. There will still be a race when user uses kubectl to forcefully delete a pod from etcd immediately.